src/UuidType.php

```diff
@@ -111,16 +100,4 @@ public function getName()
     {
         return static::NAME;
     }
-
-    /**
-     * {@inheritdoc}
-     *
-     * @param AbstractPlatform $platform
-     *
-     * @return bool
-     */
-    public function requiresSQLCommentHint(AbstractPlatform $platform)
```

Comment:
this change looks wrong to me. The `Guidtype` does not always return true, because it is mapped to the native `uuid` type of supporting platforms. But this `UuidType` is not mapped like that, and so it always requires the comment hint to avoid issues when diffing the schema (the DB introspection would detect a GuidType and so a change would be needed)

Reply:
I don't get it... are you telling that comment hint is required also on platforms that support native uuid?

Reply:
yes, because when inspecting the existing database schema, the native uuid field is mapped to a GuidType, not to the UuidType of that library, which would cause the schema diff to report a change all the time (changing the type to the same one).
Custom types that don't map to dedicated native types (not already mapped by some built-in doctrine type) require the usage of a comment hint. Custom types that can be mapped to dedicated native types require overriding `getMappedDatabaseTypes` to perform the registration (they might still require a comment hint in some platforms, if the mapping is not supported everywhere)